### ISSUE TITLE
Orchestration not being ended by terminate request

### ISSUE THREAD

**Issue by jpmartindale** [Azure/azure-functions-durable-extension#1460]
### Description
We have an orchestration that is running and our code did not naturally terminate. We are attempting to use the terminate API. Although the request is valid and returning 202 Accepted response, it is not terminating and when do a getStatus on it shows state of "Running"

> NOTE: JavaScript issues should be reported here: https://github.com/Azure/azure-functions-durable-js

### Expected behavior
We would have expected the orchestration to be terminated, go from Running -> Terminated

### Actual behavior
The orchestration is still running.

### App Details

- **Durable Functions extension version (e.g. v1.8.3)**:
- **Azure Functions runtime version (1.0 or 2.0)**: v3
- **Programming language used**: C#

### Screenshots
> If applicable, add screenshots to help explain your problem.

### If deployed to Azure
> We have access to a lot of telemetry that can help with investigations. Please provide as much of the following information as you can to help us investigate!

- **Timeframe issue observed**:
- **Function App name**: dcx-hda-func-prod
- **Function name(s)**: HighDemandAlertMonitor
- **Azure region**: East
- **Orchestration instance ID(s)**: A-8af6f8033be042ffaf4ab85331d76ef0
- **Azure storage account name**: dcxsab

> If you don't want to share your Function App or storage account name GitHub, please at least share the orchestration instance ID. Otherwise it's extremely difficult to look up information.

**Comment by ConnorMcMahon** [comment]
I'm taking a look at this now.

**Comment by ConnorMcMahon** [comment]
There are a few issues compounded together here.

1. I am seeing a lot of exceptions like the following:
```
System.ArgumentOutOfRangeException: The argument 'visibilityTimeout' is smaller than minimum of '00:00:00' (Parameter 'visibilityTimeout')
   at Microsoft.WindowsAzure.Storage.Core.Util.CommonUtility.AssertInBounds[T](String paramName, T val, T min, T max)
   at Microsoft.WindowsAzure.Storage.Queue.CloudQueue.UpdateMessageImpl(CloudQueueMessage message, TimeSpan visibilityTimeout, MessageUpdateFields updateFields, QueueRequestOptions options)
   at Microsoft.WindowsAzure.Storage.Queue.CloudQueue.UpdateMessageAsync(CloudQueueMessage message, TimeSpan visibilityTimeout, MessageUpdateFields updateFields, QueueRequestOptions options, OperationContext operationContext, CancellationToken cancellationToken)
   at Microsoft.WindowsAzure.Storage.Queue.CloudQueue.UpdateMessageAsync(CloudQueueMessage message, TimeSpan visibilityTimeout, MessageUpdateFields updateFields, QueueRequestOptions options, OperationContext operationContext)
   at DurableTask.AzureStorage.Messaging.TaskHubQueue.AbandonMessageAsync(MessageData message, SessionBase session) in C:\source\durabletask\src\DurableTask.AzureStorage\Messaging\TaskHubQueue.cs:line 238
```
This is an artifact of an integer overflow bug in our exponential backoff approach for retrying queue messages. We have fixed this in a recent release of DurableTask.AzureStorage. This should be fixed by upgrading to the latest version of the extension (v2.3.0 or soon v2.3.1). If you are still on Durable Functions v1.x, you can still get this fix by upgrading the underlying DurableTask.AzureStorage version to the latest (I believe it is 1.8.2?).

2. There is still the question of _why_ all of your terminate messages are being abandoned. It turns out that somehow your  history state got corrupted, because the DurableTask.Core code threw this exception:
```
TaskOrchestrationDispatcher-f73799b697b64c7684ae77dad4d1e70b-0: Exception while processing workItem A-8af6f8033be042ffaf4ab85331d76ef0
Exception: System.InvalidOperationException : Multiple ExecutionCompletedEvent found, potential corruption in state storage
	   at DurableTask.Core.OrchestrationRuntimeState.SetMarkerEvents(HistoryEvent historyEvent) in C:\source\durabletask\src\DurableTask.Core\OrchestrationRuntimeState.cs:line 249
   at DurableTask.Core.TaskOrchestrationDispatcher.ProcessWorkflowCompletedTaskDecision(OrchestrationCompleteOrchestratorAction completeOrchestratorAction, OrchestrationRuntimeState runtimeState, Boolean includeDetails, Boolean& continuedAsNew)
   at DurableTask.Core.TaskOrchestrationDispatcher.OnProcessWorkItemAsync(TaskOrchestrationWorkItem workItem) in C:\source\durabletask\src\DurableTask.Core\TaskOrchestrationDispatcher.cs:line 291
   at DurableTask.Core.TaskOrchestrationDispatcher.OnProcessWorkItemSessionAsync(TaskOrchestrationWorkItem workItem) in C:\source\durabletask\src\DurableTask.Core\TaskOrchestrationDispatcher.cs:line 118
   at DurableTask.Core.WorkItemDispatcher`1.ProcessWorkItemAsync(WorkItemDispatcherContext context, Object workItemObj) in C:\source\durabletask\src\DurableTask.Core\WorkItemDispatcher.cs:line 333
Inner Exception: 
```

Unfortunately, the orchestration history has been in this corrupted state since 8/15, which is as far back as our telemetry goes, so I can't diagnose why this happened. I feel like I have seen this before, but I can't find it with a quick search.

To summarize I would recommend the following actions:

1. Upgrade your dependencies so you are using our latest actions.
2. Manually delete the history for this orchestration instance in the storage table, as it is somehow corrupted.

It is a bit concerning that there is no way to more gracefully recover from a corrupted history like this. I will consider filing an issue to address this.
